lib/lightning/security.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Lightning.Security do
+  @moduledoc """
+  Security helpers.
+  """
+  alias Ecto.Changeset
+
+  @doc """
+  Redact a password inside a json on a changeset field.
+  If the field has a valid string value "password":"thesecret" becomes "password":"***".
+  """
+  @spec redact_password(Changeset.t(), atom()) :: Changeset.t()
+  def redact_password(%Changeset{valid?: true} = changeset, field) do
+    with {:ok, str_to_redact} <- Changeset.fetch_change(changeset, field),
+         true <- String.valid?(str_to_redact) do
+      redacted =
+        String.replace(
+          str_to_redact,
+          ~r/\"password\":\"\w+\"/,
+          "\\1\"password\":\"\*\*\*\""
+        )
+
+      Changeset.put_change(changeset, field, redacted)
+    else
+      _any -> changeset
+    end
+  end
```

Comment:
I missed the specification for this. while we definitely do want to scrub logs I think it's much more complicated that just looking for a password pattern - a bad actor could obfuscate this with little effort.

Take a look at `Lightning.Scrubber`, it takes a credential and looks for leaked values regardless of the key.

@taylordowns2000 care to add something here?

Reply:
There is an assertion needed for the log lines to check if any password is redacted. Please let me know if we wanna increase the scope.